### PR TITLE
Fix F.batch_norm output `variance_out` 易用性提升

### PR DESCRIPTION
**Pull request by NKNaN**
### PR Category

User Experience

### PR Types

Bug fixes

### Description

PyConvert 中 tests/test_nn_functional_batch_norm.py: _test_case_3 paddle 与 pytorch 结果不一致

经排查发现导致结果不一致的原因是 `variance_out` 的计算不一致，在更新 running_variance 时，paddle 计算使用的 variance 是biased variance，pytorch 是 unbiased variance。
image
```python
x = paddle.arange(12, dtype="float32").reshape([2, 1, 2, 3])

running_mean = paddle.to_tensor([0], dtype="float32")
running_variance = paddle.to_tensor([1], dtype="float32")
weight = paddle.to_tensor([2], dtype="float32")
bias = paddle.to_tensor([1], dtype="float32")

batch_norm_out = paddle.nn.functional.batch_norm(x, running_mean,
                                            running_variance, weight, bias, training=True)
print(batch_norm_out)
Tensor(shape=[2, 1, 2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
       [[[[-2.18650889, -1.60714364, -1.02777839],
          [-0.44841313,  0.13095212,  0.71031737]]],


        [[[ 1.28968263,  1.86904788,  2.44841313],
          [ 3.02777839,  3.60714364,  4.18650913]]]])

print(running_variance)
Tensor(shape=[1], dtype=float32, place=Place(cpu), stop_gradient=True,
       [2.09166694])

print(running_mean)
Tensor(shape=[1], dtype=float32, place=Place(cpu), stop_gradient=True,
       [0.55000013])

print(weight)
Tensor(shape=[1], dtype=float32, place=Place(cpu), stop_gradient=True,
       [2.])

print(bias)
Tensor(shape=[1], dtype=float32, place=Place(cpu), stop_gradient=True,
       [1.])
```
```python
x = torch.arange(12, dtype=torch.float32).reshape([2, 1, 2, 3])

running_mean = torch.tensor([0], dtype=torch.float32)
running_variance = torch.tensor([1], dtype=torch.float32)
weight = torch.tensor([2], dtype=torch.float32)
bias = torch.tensor([1], dtype=torch.float32)

batch_norm_out = torch.nn.functional.batch_norm(x, running_mean,
                                            running_variance, weight, bias, training=True)
print(batch_norm_out)
tensor([[[[-2.1865, -1.6071, -1.0278],
          [-0.4484,  0.1310,  0.7103]]],


        [[[ 1.2897,  1.8690,  2.4484],
          [ 3.0278,  3.6071,  4.1865]]]])

print(running_variance)
tensor([2.2000])

print(running_mean)
tensor([0.5500])

print(weight)
tensor([2.])

print(bias)
tensor([1.])
```

### CODE REVIEW

**Review by zhwesky2010 — COMMENTED** [review]
这个改动会有不兼容影响吗

**Comment by NKNaN** [comment]
> 这个改动会有不兼容影响吗

这个改动在只进行训练或者推理时一般不会影响 F.batch_norm 的返回结果（返回结果还是按照 biased variance 来算的），PaConvert 中的单测之所以返回结果不同是因为传入的 running_mean, running_variance, weight, bias 都是同一个 tensor, 所以内存是共享的， weight 和 bias 在更新完了 running_variance 之后就不一样了，导致最终返回结果不一样。 

如果在一般情况下调用了 F.batch_norm 并设置 running_variance 以及 training=True，后续操作或结果若依赖于 running_variance，则会不兼容。目前来看存量代码里没有直接遇到这种情况。但是 nn.BatchNorm1D/ nn.BatchNorm2D/ nn.BatchNorm3D 在用户应用模型的过程中会遇到这种情况，因为在训练完成之后，running_variance 记录了训练过程中的移动平均，而推理过程将直接应用 running_variance 记录的最终值进行计算，而不是用推理数据的实时方差计算，所以最终返回的结果也会改变。

**Comment by zhwesky2010** [comment]
> > 这个改动会有不兼容影响吗
> 
> 这个改动在只进行训练或者推理时一般不会影响 F.batch_norm 的返回结果（返回结果还是按照 biased variance 来算的），PaConvert 中的单测之所以返回结果不同是因为传入的 running_mean, running_variance, weight, bias 都是同一个 tensor, 所以内存是共享的， weight 和 bias 在更新完了 running_variance 之后就不一样了，导致最终返回结果不一样。
> 
> 如果在一般情况下调用了 F.batch_norm 并设置 running_variance 以及 training=True，后续操作或结果若依赖于 running_variance，则会不兼容。目前来看存量代码里没有直接遇到这种情况。但是 nn.BatchNorm1D/ nn.BatchNorm2D/ nn.BatchNorm3D 在用户应用模型的过程中会遇到这种情况，因为在训练完成之后，running_variance 记录了训练过程中的移动平均，而推理过程将直接应用 running_variance 记录的最终值进行计算，而不是用推理数据的实时方差计算，所以最终返回的结果也会改变。

这个不兼容影响面大吗？一般不是直接用batch_norm，而是用nn.BatchNorm1D/ nn.BatchNorm2D/ nn.BatchNorm3D这种形式。

**Comment by NKNaN** [comment]
> > > 这个改动会有不兼容影响吗
> > 
> > 
> > 这个改动在只进行训练或者推理时一般不会影响 F.batch_norm 的返回结果（返回结果还是按照 biased variance 来算的），PaConvert 中的单测之所以返回结果不同是因为传入的 running_mean, running_variance, weight, bias 都是同一个 tensor, 所以内存是共享的， weight 和 bias 在更新完了 running_variance 之后就不一样了，导致最终返回结果不一样。
> > 如果在一般情况下调用了 F.batch_norm 并设置 running_variance 以及 training=True，后续操作或结果若依赖于 running_variance，则会不兼容。目前来看存量代码里没有直接遇到这种情况。但是 nn.BatchNorm1D/ nn.BatchNorm2D/ nn.BatchNorm3D 在用户应用模型的过程中会遇到这种情况，因为在训练完成之后，running_variance 记录了训练过程中的移动平均，而推理过程将直接应用 running_variance 记录的最终值进行计算，而不是用推理数据的实时方差计算，所以最终返回的结果也会改变。
> 
> 这个不兼容影响面大吗？一般不是直接用batch_norm，而是用nn.BatchNorm1D/ nn.BatchNorm2D/ nn.BatchNorm3D这种形式。

如果算nn.BatchNorm1D/ nn.BatchNorm2D/ nn.BatchNorm3D的话影响面是比较大的

**Comment by zhwesky2010** [comment]
> > > > 这个改动会有不兼容影响吗
> > > 
> > > 
> > > 这个改动在只进行训练或者推理时一般不会影响 F.batch_norm 的返回结果（返回结果还是按照 biased variance 来算的），PaConvert 中的单测之所以返回结果不同是因为传入的 running_mean, running_variance, weight, bias 都是同一个 tensor, 所以内存是共享的， weight 和 bias 在更新完了 running_variance 之后就不一样了，导致最终返回结果不一样。
> > > 如果在一般情况下调用了 F.batch_norm 并设置 running_variance 以及 training=True，后续操作或结果若依赖于 running_variance，则会不兼容。目前来看存量代码里没有直接遇到这种情况。但是 nn.BatchNorm1D/ nn.BatchNorm2D/ nn.BatchNorm3D 在用户应用模型的过程中会遇到这种情况，因为在训练完成之后，running_variance 记录了训练过程中的移动平均，而推理过程将直接应用 running_variance 记录的最终值进行计算，而不是用推理数据的实时方差计算，所以最终返回的结果也会改变。
> > 
> > 
> > 这个不兼容影响面大吗？一般不是直接用batch_norm，而是用nn.BatchNorm1D/ nn.BatchNorm2D/ nn.BatchNorm3D这种形式。
> 
> 如果算nn.BatchNorm1D/ nn.BatchNorm2D/ nn.BatchNorm3D的话影响面是比较大的

那这个要不改一下paconvert的单测，是不是这个单测写法不太合理，改成合理写法会有问题吗

**Comment by NKNaN** [comment]
> > > > > 这个改动会有不兼容影响吗
> > > > 
> > > > 
> > > > 这个改动在只进行训练或者推理时一般不会影响 F.batch_norm 的返回结果（返回结果还是按照 biased variance 来算的），PaConvert 中的单测之所以返回结果不同是因为传入的 running_mean, running_variance, weight, bias 都是同一个 tensor, 所以内存是共享的， weight 和 bias 在更新完了 running_variance 之后就不一样了，导致最终返回结果不一样。
> > > > 如果在一般情况下调用了 F.batch_norm 并设置 running_variance 以及 training=True，后续操作或结果若依赖于 running_variance，则会不兼容。目前来看存量代码里没有直接遇到这种情况。但是 nn.BatchNorm1D/ nn.BatchNorm2D/ nn.BatchNorm3D 在用户应用模型的过程中会遇到这种情况，因为在训练完成之后，running_variance 记录了训练过程中的移动平均，而推理过程将直接应用 running_variance 记录的最终值进行计算，而不是用推理数据的实时方差计算，所以最终返回的结果也会改变。
> > > 
> > > 
> > > 这个不兼容影响面大吗？一般不是直接用batch_norm，而是用nn.BatchNorm1D/ nn.BatchNorm2D/ nn.BatchNorm3D这种形式。
> > 
> > 
> > 如果算nn.BatchNorm1D/ nn.BatchNorm2D/ nn.BatchNorm3D的话影响面是比较大的
> 
> 那这个要不改一下paconvert的单测，是不是这个单测写法不太合理，改成合理写法会有问题吗

好的，改成合理写法没有问题的
